src/components/VtkThreeView.vue

```diff
@@ -326,27 +326,33 @@ export default defineComponent({
           light.setPositional(false);
         }
 
-        const dims = image.getDimensions();
-        const spacing = image.getSpacing();
-        const spatialDiagonal = vec3.length(
-          vec3.fromValues(
-            dims[0] * spacing[0],
-            dims[1] * spacing[1],
-            dims[2] * spacing[2]
-          )
-        );
+        if (animating) {
+          mapper.setSampleDistance(0.75);
+          mapper.setMaximumSamplesPerRay(1000);
+          mapper.setGlobalIlluminationReach(0);
+        } else {
+          const dims = image.getDimensions();
+          const spacing = image.getSpacing();
+          const spatialDiagonal = vec3.length(
+            vec3.fromValues(
+              dims[0] * spacing[0],
+              dims[1] * spacing[1],
+              dims[2] * spacing[2]
+            )
+          );
 
-        // Ensure that the maximum samples per ray are not exceeded.
-        let sampleDistance = mapper.getSampleDistance();
-        if (
-          spatialDiagonal / sampleDistance >
-          mapper.getMaximumSamplesPerRay()
-        ) {
-          sampleDistance =
-            spatialDiagonal / (mapper.getMaximumSamplesPerRay() - 1);
+          // Use the average spacing for sampling by default
+          let sampleDistance = spacing.reduce((a, b) => a + b) / 3.0;
+          // Adjust the volume sampling by the quality slider value
+          sampleDistance /= params.volumeQuality;
+          const samplesPerRay = spatialDiagonal / sampleDistance + 1;
+          mapper.setMaximumSamplesPerRay(samplesPerRay);
+          mapper.setSampleDistance(sampleDistance);
+          // Adjust the global illumination reach by volume quality slider
+          mapper.setGlobalIlluminationReach(
+            enabled ? 0.2 * params.volumeQuality : 0
+          );
```

Comment:
Are you also going to add more secondary rays?

Reply:
Yes, indirect illumination. That will be a vtk-js change though.